Fixes flakey integration test/syncs deps

aittalam opened the pull request:
## What's changing

I updated pyproject.toml to use more recent libs. The main reason was transformers so we could run more recent models but I ran a general update using `pip-review` to get latest versions. These are the steps I followed:

- used pip-review to update all libs
- check there were no inconsistencies
- test lm-buddy with the new set of libs
- updated `pyproject.toml` by picking the updated versions from the output of `pip freeze`
- re-installed everything in a new env and made sure there were no inconsistencies
- tested lm-buddy again

## How to test it

The tests I ran were unit/integration tests + manually running lm-buddy for summarization eval (currently our main job).
Interestingly, `tests/integration/test_lm_harness.py::test_lm_harness_job` passes when I test locally

aittalam commented on pyproject.toml:
```diff
@@ -26,47 +26,45 @@ dependencies = [
     "protobuf>=3.20.2",
     "urllib3>=1.26.18,<2",
-    "pydantic>=2.6.4",
-    "pydantic-yaml>=1.2.0",
```
I kept only the == as there was redundancy here and having both would have resulted in having ==, but am happy to put >= if the intention was to keep them updated

binaryaaron replied:
yeah, i changed it back to make sure it worked via pip-compile.

aittalam commented on pyproject.toml:
```diff
@@ -26,47 +26,45 @@ dependencies = [
     "ray[default]==2.30.0",
     "loguru==0.7.2",
-    "s3fs",
+    "s3fs==2024.5.0",
```
pinned to what we are actually running rn

aittalam commented on pyproject.toml:
```diff
@@ -26,47 +26,45 @@ dependencies = [
 
 [project.optional-dependencies]
-ruff = ["ruff==0.2.1"]
+ruff = ["ruff==0.5.6"]
```
@binaryaaron is this ok? Not sure whether this has to be changed anywhere else for testing on github

binaryaaron replied:
made it match what's being used internally for now

aittalam commented:
Two questions for @binaryaaron: 

1) ruff on github returns an error which I do not get at precommit nor when I run it manually. Is there anything else I should change in the repo to make sure the two are aligned? (nothing major, I'd just like to prevent errors happening here if I can)

2) I still wonder what is different between my local tests and github's as they all pass locally (in particular the lm-harness one which breaks online):
<img width="875" alt="image" src="https://github.com/user-attachments/assets/11568d86-de2b-479a-bc05-5f140d79e5ef">

binaryaaron commented:
I added some minor changes and changed the test task to one that doesn't require HF remote code. there's something odd with how those args are being passed; i opened #115 to revisit this after the demo.

aittalam commented:
> I added some minor changes and changed the test task to one that doesn't require HF remote code. there's something odd with how those args are being passed; i opened #115 to revisit this after the demo.

That's great, many thanks!
